mesa/visualization/templates/js/GridDraw.js

```diff
@@ -21,80 +21,98 @@ time. Thus, it should be possible to turn different layers on and off.
    ],
  1:[
-	{"Shape": "circle", "x": 0, "y": 0, "r": 0.5, "Color": "#AAAAAA", "Filled": "true", "Layer": 1, "text": 'A', "text_color": "white"},
-	{"Shape": "circle", "x": 1, "y": 1, "r": 0.5, "Color": "#AAAAAA", "Filled": "true", "Layer": 1, "text": 'B', "text_color": "white"}
-	{"Shape": "arrowHead", "x": 1, "y": 0, "heading_x": -1, heading_y: 0, "scale": 0.5, "Color": "green", "Filled": "true", "Layer": 1, "text": 'C', "text_color": "white"}
+        {"Shape": "circle", "x": 0, "y": 0, "r": 0.5, "Color": ["#00aa00", "#aa00aa"], "stroke_color": "red", "Filled": "true", "Layer": 1, "text": 'A', "text_color": "white"},
+        {"Shape": "circle", "x": 1, "y": 1, "r": 0.5, "Color": ["#00aa00", "#aa00aa"], "stroke_color": "red", "Filled": "true", "Layer": 1, "text": 'B', "text_color": "white"}
+        {"Shape": "arrowHead", "x": 1, "y": 0, "heading0": -1, heading1: 0, "scale": 0.5, "Color": ["#00aa00", "#aa00aa"], "stroke_color": "red", "Filled": "true", "Layer": 1, "text": 'C', "text_color": "white"}
    ]
 }
 
 */
 
 var GridVisualization = function(width, height, gridWidth, gridHeight, context) {
 
-	// Find cell size:
-	var cellWidth = Math.floor(width / gridWidth);
-	var cellHeight = Math.floor(height / gridHeight);
+        // Find cell size:
+        var cellWidth = Math.floor(width / gridWidth);
+        var cellHeight = Math.floor(height / gridHeight);
 
         // Find max radius of the circle that can be inscribed (fit) into the
         // cell of the grid.
-	var maxR = Math.min(cellHeight, cellWidth)/2 - 1;
+        var maxR = Math.min(cellHeight, cellWidth)/2 - 1;
 
-	// Calls the appropriate shape(agent)
+        // Calls the appropriate shape(agent)
         this.drawLayer = function(portrayalLayer) {
-		for (var i in portrayalLayer) {
-			var p = portrayalLayer[i];
+                for (var i in portrayalLayer) {
+                        var p = portrayalLayer[i];
+
+                        // If p.Color is a string scalar, cast it to an array.
+                        // This is done to maintain backwards compatibility
+                        if (!Array.isArray(p.Color))
+                                p.Color = [p.Color];
+
                         // Does the inversion of y positioning because of html5
                         // canvas y direction is from top to bottom. But we
                         // normally keep y-axis in plots from bottom to top.
                         p.y = gridHeight - p.y - 1;
-			if (p.Shape == "rect")
-				this.drawRectangle(p.x, p.y, p.w, p.h, p.Color, p.Filled, p.text, p.text_color);
-			else if (p.Shape == "circle")
-				this.drawCircle(p.x, p.y, p.r, p.Color, p.Filled, p.text, p.text_color);
+
+                        // If the stroke color is not defined, then the first color in the colors array is the stroke color.
+                        if (!p.stroke_color)
+                                p.stroke_color = p.Color[0]
```

Comment:
This line should be inline with the `if` statement above it. However, it would seem that is the pattern that has been followed here, so don't worry about it in this PR.